### PR TITLE
corrected various bug on movie creation

### PR DESCRIPTION
**Pull request by phpgeekfr**
corrected various bug on movie creation and updated rules

### CODE REVIEW

File: libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.html [hunk 1]
```diff
@@ -28,10 +28,10 @@
       </ng-template>
     </button>
     <button mat-stroked-button color="primary" [disabled]="!selection.selected.length || !!processedTitles"
-      (click)="submitSelectedMovies()">
+      (click)="submitOrPublishSelectedMovies()">
       <ng-container *ngIf="submittedTitles === 0 else submitinprogress">
         <mat-icon svgIcon="archipel_content"></mat-icon>
-        <span>Publish to Marketplace</span>
+        <span>{{currentApp === 'festival' ? 'Publish to Marketplace': 'Submit to Archipel Content'}}</span>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.html:34]: I think this is not sustainable. Most of our apps will have a movie form. Let's keep it neutral (I validated it with @Jackseed last Friday).

File: libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.html [hunk 2]
```diff
@@ -28,10 +28,10 @@
       </ng-template>
     </button>
     <button mat-stroked-button color="primary" [disabled]="!selection.selected.length || !!processedTitles"
-      (click)="submitSelectedMovies()">
+      (click)="submitOrPublishSelectedMovies()">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.html:31]: Would be nice to have a shorter, less app dependant method name here.

Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.html:31]: Same goes for he other methods below

File: libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.html [hunk 3]
```diff
@@ -125,10 +125,10 @@
       <ng-container *ngIf="element.movie.id; else create">
         <button mat-icon-button color="accent" [disabled]="isSaveOrUpdateDisabledForTitle(element)"
           (click)="updateMovie(element)"><mat-icon svgIcon="access_time"></mat-icon></button>
-        <button mat-stroked-button color="primary" [disabled]="!isTitleValidForSubmit(element)"
-          (click)="submitMovie(element)">
+        <button mat-stroked-button color="primary" [disabled]="!isTitleValidForSubmitOrPublish(element)"
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.html:128]: Whenever you use a method for an input you should be able to use a Pipe. Which is much more efficient for performance, and take advange of injection dependancy.

File: libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.ts [hunk 4]
```diff
@@ -72,11 +73,11 @@ export class TableExtractedMoviesComponent implements OnInit {
...
- * @dev Once created (as draft), titls can be submitted to Archipel Content
- */
-  async submitMovie(importState: MovieImportState): Promise<boolean> {
-    await this.submitToArchipelContent(importState);
-    this.snackBar.open('Title submitted !', 'close', { duration: 3000 });
+    * @dev Once created (as draft), depending on the app, titles can be submitted to Archipel Content or accepted (ndlr: published) directly
+   */
+  async submitOrPublishMovie(importState: MovieImportState): Promise<boolean> {
+    await this.submitOrPublishToArchipelContent(importState);
+    this.snackBar.open(`Title ${this.currentApp === 'catalog' ? 'submitted' : 'published'} !`, 'close', { duration: 3000 });
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.ts:80]: Same as above try to keep things neutral

File: libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.ts [hunk 5]
```diff
@@ -99,18 +100,18 @@ export class TableExtractedMoviesComponent implements OnInit {
...
    */
-  async submitSelectedMovies(): Promise<boolean> {
+  async submitOrPublishSelectedMovies(): Promise<boolean> {
     try {
-      const creations = this.selection.selected.filter(importState => importState.movie.id && !hasImportErrors(importState) && this.isTitleValidForSubmit(importState));
+      const creations = this.selection.selected.filter(importState => importState.movie.id && !hasImportErrors(importState) && this.isTitleValidForSubmitOrPublish(importState));
       for (const movie of creations) {
         this.submittedTitles++;
-        await this.submitToArchipelContent(movie);
+        await this.submitOrPublishToArchipelContent(movie);
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.ts:108]: As @lsenta putted out, we shouldn't use "product name" in the code.

File: libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.ts [hunk 6]
```diff
@@ -170,15 +166,15 @@ export class TableExtractedMoviesComponent implements OnInit {
    * Change title status from 'Draft' to 'Submitted to Archipel Content'
    * @param importState
    */
-  private async submitToArchipelContent(importState: MovieImportState): Promise<boolean> {
+  private async submitOrPublishToArchipelContent(importState: MovieImportState): Promise<boolean> {
     const data = this.rows.data;
-    importState.movie.main.storeConfig.status = 'submitted';
+    importState.movie.main.storeConfig.status = this.currentApp === 'catalog' ? 'submitted' : 'accepted';
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.ts:171]: This should be managed in a centralized place (like the service).

File: libs/utils/src/lib/helpers.ts [hunk 7]
```diff
@@ -12,6 +12,33 @@ export function cleanModel<T>(data: T): T {
   return JSON.parse(JSON.stringify(data));
 }
 
+export function isObject(item) {
+  return (item && typeof item === 'object' && !Array.isArray(item));
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/helpers.ts:16]: Maybe you want to check `!null` too. Yes null is an object ...

File: libs/utils/src/lib/helpers.ts [hunk 8]
```diff
@@ -12,6 +12,33 @@ export function cleanModel<T>(data: T): T {
...
+}
+
+/**
+ * @dev this method is used to deeply merge two object without loosing data
+ * Use this instead of { ...this.query.getActive(), ...this.form.value }
+ * @param target 
+ * @param source 
+ */
+export default function mergeDeep(target, source) {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/helpers.ts:25]: Why would you export this as default ? It's very confusing. Please export it like the rest of the app ^^

Comment (phpgeekfr) [reply libs/utils/src/lib/helpers.ts:25]: sry copy paste ^^

File: libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.ts [hunk 9]
```diff
@@ -170,15 +164,15 @@ export class TableExtractedMoviesComponent implements OnInit {
    * Change title status from 'Draft' to 'Submitted to Archipel Content'
    * @param importState
    */
-  private async submitToArchipelContent(importState: MovieImportState): Promise<boolean> {
+  private async publish(importState: MovieImportState): Promise<boolean> {
     const data = this.rows.data;
-    importState.movie.main.storeConfig.status = 'submitted';
+    importState.movie.main.storeConfig.status = getMoviePublishStatus(this.currentApp);
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/import/table-extracted-elements/movies/movies.component.ts:169]: Maybe add the todo of the issue you create here ? Or it's unrelated ?

**Pull request merged**